### DISCUSSION TITLE
Scope to utility classes to aid the opt-in typography system

### DISCUSSION THREAD

**Discussion by mildlyfunctional** [tailwindlabs/tailwindcss#3890] [Ideas]
Opt-in typography in combination with utility classes could be a love match.

But the content inside the typography class `.prose` is not only CMS or plain Markdown but also MDX or MDsvex where 
components can be used inside markdown.

This means that utility classes need a (preferably slightly) higher specificity than the definitions of the typography classes.

For the sake of simplicity I'm using h1 as an example but think of more complex component structures:

Say we have `.prose h1`, I want to be able to override the side with `.text-5xl` because in this instance, bigger is better.
It is not possible, because .prose h1 has more weight.

With normal tailwind (not jit compiler) the important config can be misused for this purpose:
```
{
  important: '.tw',
  ...
}
```

Now if we have `.prose h1` it can be and we use the `.text-5xl` the class definition is `.tw .text-5xl`, so enough specificity.

I don't think the `important` option should be used for this. The [expected behaviour](https://github.com/tailwindlabs/tailwindcss-typography/issues/152#issue-831688692) is different.

I do think it should be possible set a scope class to the utility classes alone for this purpose.

**Comment by adamwathan** [comment]
Hey! This is the whole reason we added the ability to use a selector for "important", to increase the specificity without going so high as to defeat inline styles. No plans to change this at least not without a compelling argument for why. I would recommend using an ID not a class though, it doesn't add enough specificity to defeat complex multi-class selectors otherwise. Originally we actually threw an exception when you used a class, not sure what happened there but never intended for that validation to be removed :/

**Reply by adamwathan** [reply]
The link you shared to the typography issue is unrelated, the important option does not apply to component classes like prose, because there's no reason to ever increase the specificity of component classes.

**Reply by mildlyfunctional** [reply]
Ok, so it is safe to use it for this purpose? Happy to hear that. 

When extracting components to a plugin, like `.btn` (from [this example](https://tailwindcss.com/docs/extracting-components#writing-a-component-plugin)), than I think that component should be able to 'inherit' the scope class from the important option. So it depends I think.

**Reply by mildlyfunctional** [reply]
And also. Pleas leave the choice of the selector to the user for more fine-grained control (and advice it to be an id) :)
